code/datums/components/material_container.dm

```diff
@@ -43,14 +45,17 @@
 			materials[id] = new mat_path()
 
 /datum/component/material_container/proc/OnExamine(mob/user)
-	for(var/I in materials)
-		var/datum/material/M = materials[I]
-		var/amt = amount(M.id)
-		if(amt)
-			to_chat(user, "<span class='notice'>It has [amt] units of [lowertext(M.name)] stored.</span>")
+	if(show_on_examine)
+		for(var/I in materials)
+			var/datum/material/M = materials[I]
+			var/amt = amount(M.id)
+			if(amt)
+				to_chat(user, "<span class='notice'>It has [amt] units of [lowertext(M.name)] stored.</span>")
 
 /datum/component/material_container/proc/OnAttackBy(obj/item/I, mob/living/user)
 	var/list/tc = allowed_typecache
+	if(disable_attackby)
```

Comment:
For clarity; there may be instances where we want something to hold materials and it may have some other method accepting materials (ie: taking them in in `process` or something), but we don't want players themselves to be able to put object into the container willy-nilly.

Reply:
Is that why you put the check here instead of just disabling the signal registration?